src/Features/Core/Portable/ConvertAnonymousTypeToTuple/AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.ConvertAnonymousTypeToTuple
+{
+    internal abstract class AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer<
+        TSyntaxKind,
+        TAnonymousObjectCreationExpressionSyntax> 
+        : AbstractCodeStyleDiagnosticAnalyzer
+        where TSyntaxKind : struct
+        where TAnonymousObjectCreationExpressionSyntax : SyntaxNode
+    {
+        protected AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.ConvertAnonymousTypeToTupleDiagnosticId,
+                   new LocalizableResourceString(nameof(FeaturesResources.Convert_to_tuple), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+                   new LocalizableResourceString(nameof(FeaturesResources.Convert_to_tuple), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+        {
+        }
+
+        protected abstract TSyntaxKind GetAnonymousObjectCreationExpressionSyntaxKind();
+        protected abstract int GetInitializerCount(TAnonymousObjectCreationExpressionSyntax anonymousType);
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
```

Comment:
>SemanticSpanAnalysis [](start = 42, length = 20)

Can this be a syntax analyzer only (no semantics)?

Reply:
most likely!  thanks!